website/archiver/utils.py

```diff
@@ -221,14 +221,22 @@ def find_registration_files(values, node):
         ret.append(find_registration_file(values['extra'][i], node) + (i,))
     return ret
 
-def find_question(schema, qid):
+def get_title_for_question(schema, path):
+    path = path.split('.')
+    root = path.pop(0)
+    item = None
     for page in schema['pages']:
         questions = {
             q['qid']: q
             for q in page['questions']
         }
-    if qid in questions:
-        return questions[qid]
+        if root in questions:
+            item = questions[root]
+    title = item.get('title')
+    while len(path):
```

Comment:
Won't this be an infinite loop? I don't see anything that changes the length of path.